src/AutoRest.CSharp/Common/Output/Models/Types/EnumType.cs

```diff
@@ -55,15 +55,21 @@ public EnumType(InputEnumType input, string defaultNamespace, string defaultAcce
             }
 
             ValueType = typeFactory.CreateType(input.EnumValueType);
+            IsStringValueType = ValueType.Equals(typeof(string));
+            IsIntValueType = ValueType.Equals(typeof(int)) || ValueType.Equals(typeof(long));
+
             Description = input.Description;
             IsExtensible = isExtensible;
         }
 
         public CSharpType ValueType { get; }
         public bool IsExtensible { get; }
-        public bool IsStringValueType => ValueType.Equals(typeof(string));
-        public bool IsIntValueType => ValueType.Equals(typeof(Int32)) || IsLongValueType;
-        public bool IsLongValueType => ValueType.Equals(typeof(Int64));
+        public bool IsStringValueType { get; }
+        public bool IsIntValueType { get; }
```

Comment:
No `IsFloatValueType` ? Do we still need special handle for Float?

Reply:
We do not have that flag because we do not need it.
I only add a flag when it is necessary not because there is such a case.

IsStringValueType flag is used when we want to know if we need the extra method
IsIntValueType flag is used in fixed enum's serialization to determine if we need to use cast instead of calling `ToSerialXX` method.

In summary, for fixed enums, when IsIntValueType is true, we have one case, and when it is false, we have the other case (string and float make no difference).
For extensible enums, when IsStringValueType is true, we have one case, and when it is false, we have the other case (int and float make no difference).
Therefore we do not need any more flags. We could add them when we need some.